### PR TITLE
Upgrade to Xtext 2.19, LSP4J 0.8.0 and fix LSP server

### LINKED ISSUE #1030: 2.5.0.M2/M3 LSP does not work with VSC
It seems that from 2.5.2.M2 on the VSC cooperation with LSP doe snot work any more.
According to @Confectrician nothing has been changed on VSC side. see open issue: [issue 141](https://github.com/openhab/openhab-vscode/issues/141 )

### PR DESCRIPTION
**Pull request by wborn**
Every connection to the LSP server failed because it always threw a `org.eclipse.smarthome.model.ide.SitemapIdeSetup NoClassDefFoundError`.
Which is caused by the `SitemapIdeSetup` residing in a package that is split between both the `org.openhab.core.model.item.ide` and the `org.openhab.core.model.sitemap.ide` bundles.
The `SitemapIdeSetup` class is also using classes from the `org.eclipse.smarthome.model` package which is also a package split between the `org.openhab.core.model.items` and `org.openhab.core.model.sitemap` bundles.
After renaming the packages used for the sitemap classes the NoClassDefFoundError is resolved and LSP works properly again.
Many imports were updated for the sitemap model package changes.

Fixes #1030

---

For the Xtext update we need to deploy [io.github.classgraph_4.8.35.v20190528-1517.jar](https://download.eclipse.org/tools/orbit/downloads/drops2/I20191011000200/repository/plugins/io.github.classgraph_4.8.35.v20190528-1517.jar) from Eclipse Orbit (see [recipe](https://git.eclipse.org/c/orbit/orbit-recipes.git/tree/github/classgraph/io.github.classgraph_4.8.35)) to our own repo because feature validation fails with the version available from Maven Central.

If this is the way to go I can also make some PRs with the changes for the other repos:

openhab2-addons: https://github.com/wborn/openhab2-addons/commit/72fc184b33aff7c1f4ec8fc452b76b4d039a9321
openhab-webui: https://github.com/wborn/openhab-webui/commit/15e76b14a075d1231eec4e045642dba36ff972fb

I've renamed the sitemap model packages similarly to how they are also used with the thing and script models. As you can see in the webui changes it does result in `sitemap.sitemap` being used in the imports which does seem a bit odd.

My local build with these changes is available in this release:

https://github.com/wborn/openhab-core/releases/tag/update-xtext-lsp4j-fix-ncdfe-release-tag

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Wow, great work, thanks!

> For the Xtext update we need to deploy io.github.classgraph_4.8.35.v20190528-1517.jar from Eclipse Orbit to our own repo

@J-N-K You're the master of our dependency management: Shall this be added to https://bintray.com/openhab/mvn/openhab-addons-deps (despite the fact that it isn't an add-on dependency, but a core dependency)? Or do you have any better place?

**Comment by wborn** [comment]
We might also want to use this repo for the artifacts currently obtained from @maggu2810's private repo (https://github.com/openhab/openhab-core/issues/960).

**Comment by J-N-K** [comment]
I think that creating an additional repo for a single dependency is a bit overkill. Can we rename the repo to openhab-dep? Would be a better fit.

**Comment by kaikreuzer** [comment]
I would just leave the name as is and don't worry too much about it.
@J-N-K Could you take care of uploading the new lib there?

**Comment by J-N-K** [comment]
Done. Please try if it works as expected.

**Comment by wborn** [comment]
Many thanks @J-N-K! I think it works after updating the GAV and adding an exclude of the original artifact.

File: itests/org.openhab.core.thing.xml.tests/itest.bndrun [hunk 1]
```diff
@@ -46,4 +46,5 @@ Fragment-Host: org.openhab.core.thing.xml
 	org.apache.servicemix.specs.jaxb-api-2.2;version='[2.9.0,2.9.1)',\
 	org.apache.servicemix.specs.stax-api-1.2;version='[2.9.0,2.9.1)',\
 	tec.uom.lib.uom-lib-common;version='[1.0.3,1.0.4)',\
-	tec.uom.se;version='[1.0.10,1.0.11)'
+	tec.uom.se;version='[1.0.10,1.0.11)',\
+	org.openhab.core.storage.json;version='[2.5.0,2.5.1)'
```
[the remark this answers is not in the document]
Comment (wborn) [reply itests/org.openhab.core.thing.xml.tests/itest.bndrun]: Yes it would be nice to have some reusable configurations like that.

**Comment by maggu2810** [comment]
Just to track it somewhere:

With this change openHAB relies on "Log Service Specification" version 1.4.

**Comment by wborn** [comment]
Do you think that is a positive change or might it cause problems?

**Comment by maggu2810** [comment]
It might cause problems.

The "Equinox Common" tries to access the Equinox Logger (see package "org.eclipse.equinox.log") that one is provided by "Equinox Supplement". Additionally if you use the Equinox framework it is also provided by that one.

The Equinox Logger implements the Logger interface of the OSGi Log
Service specification 1.4
https://github.com/eclipse/rt.equinox.framework/blob/63a9e1075ab029c5090a3d50cf52b82f052c62f6/bundles/org.eclipse.osgi/supplement/src/org/eclipse/equinox/log/Logger.java#L23

The "Equinox Supplement" does not usage version constraints at all on
its imports:
https://github.com/eclipse/rt.equinox.framework/blob/63a9e1075ab029c5090a3d50cf52b82f052c62f6/bundles/org.eclipse.osgi/supplement/META-INF/MANIFEST.MF#L25

Pax Logging provides the OSGi Log Service specification 1.3.

So, as soon as "Equinox Common" is using "Equinox Supplement" that requires "org.osgi.service.log" (without a version constraint) and it is bound to Pax Logging that provides 1.3 and it accesses the Logger interface that class is only available on 1.4 you will get a CNFE.

If you are using the Equinox Framework (which you did for the openHAB Distribution) it works because this one bundles the desired stuff.

If you use Apache Felix and so no other bundle that provides an 1.4 log implementation. Boom :wink: 

In general I assume a core framework should not provide a service implementation of the compendium itself, but that's just my feeling.

**Comment by wborn** [comment]
I think you're right about that and I only did some integration testing with Equinox at the time. It might be  good to track such incompatibilities in issues so they are also more easily discovered by others that use OHC without Equinox.

**Comment by maggu2810** [comment]
Just FYI:
For the ones like me that want to use the Felix framework in the current Karaf versions and the recent code base, this [org.eclipse.equinox.supplement](https://bintray.com/maggu2810/maven/org.eclipse.equinox.supplement#files/de%2Fmaggu2810%2Fthirdparty%2Fmodified%2Forg%2Feclipse%2Fplatform%2Forg.eclipse.equinox.supplement%2F1.9.0) should work.

**Pull request merged**